### DISCUSSION TITLE
Post several transactions via API without grouping

### DISCUSSION THREAD

**Discussion by Londoneye02** [firefly-iii/firefly-iii#7477] [Firefly III]
Hi. 

I am trying to parse some transactions from a Google Sheet to Firefly using Node Red. Each transaction is one line. I am able to get all the transactions as an array and need to send them to the Firefly API.

My problem is that if I try to send more than one transaction I am forced to give a "group title", otherwise the POST won't work
```
`{
    "error_if_duplicate_hash": false,
    "apply_rules": false,
    "group_title": "mygroup",
    "transactions": [
        {
            "type": "withdrawal",
            "source_id": "227",
            "destination_id": "117",
            "date": "2023-02-24T10:39:47+01:00",
            "description": "Test1",
            "amount": "210"
        },
         {
            "type": "withdrawal",
            "source_id": "227",
            "destination_id": "117",
            "date": "2023-02-24T10:39:47+01:00",
            "description": "Test2",
            "amount": "210"
        }
       
    ]
}	`
```
I don't want to group the transactions as they have nothing to do with each other, and it is not simple to manage them that way.
![image](https://user-images.githubusercontent.com/29100907/236949303-043cf638-28aa-4002-8ba8-43fcd21c69b8.png)

Is it possible to do it as I need?

Thank you in advance

**Comment by JC5** [comment]
You'll need to submit them separately.

**Reply by coolcow** [reply]
That's how it is, but in case constructive suggestions are desired on this topic:

It would be nice to have the possibility to import multiple transactions with a single POST, and without having to group them.

Why? Importing thousands of transactions via the API seperatly works, but feels like a workaround with slow performance that produces unnecessary load on the server, network and client.

Also: currently when `error_if_duplicate_hash` is set to `true` while group-importing multiple transactions, the api aborts on the first duplicate it finds. The transactions before are imported, the ones after are not.
IMHO it would be a big improvement if the api would import all transactions that are not duplicates, and then abort at the end with all failed transactions indices in the error message.

Both things combined would allow to write an importer that import a bunch transactions at onces, with the possibility to react properly on the skipped duplicates.
